### PR TITLE
Add docs for creating custom lemmy frontend

### PR DESCRIPTION
**Pull request by Nutomic**
I think there is a lot of potential with custom frontends for Lemmy, so I decided to write a guide. 

@IronOxidizer Please have a look and tell me if anything is missing.

### CODE REVIEW

File: docs/src/contributing/custom_frontend.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  if (headers['x-real-ip']) {
+    out['x-real-ip'] = headers['x-real-ip'];
+  }
+  if (headers['x-forwarded-for']) {
+    out['x-forwarded-for'] = headers['x-forwarded-for'];
+  }
+
+  return out;
+}
+```
```
Comment (Nutomic) [inline docs/src/contributing/custom_frontend.md:62]: @dessalines this is probably incomplete

File: docs/src/contributing/custom_frontend.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Creating a Custom Frontend
+
+The backend and frontend are completely decoupled, and run in independent Docker containers. They only communicate over the [Lemmy API](websocket_http_api.md), which makes it quite easy to write alternative frontends.
+
+## Development
+
+You can use any language to create a custom frontend. The easiest option would be forking our [official frontend](https://github.com/LemmyNet/lemmy-ui), [lemmy-lite](https://github.com/IronOxidizer/lemmy-lite), or the [lemmy-frontend-example](https://github.com/LemmyNet/lemmy-front-end-example). In any case, the principle is the same: bind to `LEMMY_EXTERNAL_HOST` (default: `localhost:8536`) and handle requests using the Lemmy API at `LEMMY_INTERNAL_HOST` (default: `lemmy:8536`). Also use `LEMMY_HTTPS` to generate links with the correct protocol.
+
+To test, follow the [local development instructions](local_development.md) (you only need the database and backend dependencies). Then do `cargo run` in the Lemmy repo. Start your frontend, using `localhost:8536` for the API, and serving requests on another port.
```
Comment (dessalines) [inline docs/src/contributing/custom_frontend.md:9]: Here I would link to docker development instead... it does run the lemmy-ui (which would be unecessary for someone developing an alternate front end), but it also sets up pictrs and iframely too, which alternate front ends might need to make sure images are being rendered.

File: docs/src/contributing/custom_frontend.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+To test, follow the [local development instructions](local_development.md) (you only need the database and backend dependencies). Then do `cargo run` in the Lemmy repo. Start your frontend, using `localhost:8536` for the API, and serving requests on another port. 
+
+## Deploy with Docker
+
+The next step is building a Docker image from your frontend. If you forked an existing project, it should already include a `Dockerfile` and instructions to build it. Otherwise, try searching for your language on [dockerhub](https://hub.docker.com/), official images usually have build instructions in their readme. After building the Docker image, you need to push it to a Docker registry (such as [dockerhub](https://hub.docker.com/)). Then update the `docker-compose.yml` on your server to reference your image. Look for the following section:
```
Comment (dessalines) [inline docs/src/contributing/custom_frontend.md:13]: Add a space after this line, I've seen issues with commonmark without it.

File: docs/src/contributing/custom_frontend.md (hunk 1 above)
Comment (dessalines) [reply docs/src/contributing/custom_frontend.md:62]: Just add these two lines to the bottom.

```
let headers = setForwardedHeaders(req.headers);
let client = new LemmyHttp(httpUri, headers);
```

**Review by dessalines — CHANGES_REQUESTED** [review]
Just those above.

**Comment by Nutomic** [comment]
Updated and force pushed with your suggestions @dessalines. I also extended the introduction a bit.

**Comment by dessalines** [comment]
Looks good to me.

**Pull request merged**